generator/graphql-kotlin-schema-generator/src/main/kotlin/com/expediagroup/graphql/generator/internal/types/utils/functionReturnTypes.kt

```diff
@@ -45,6 +45,7 @@ import kotlin.reflect.full.createType
  */
 internal fun getWrappedReturnType(returnType: KType): KType {
     return when {
```

Comment:
Can we have a better recursive call here? Would this work?

```kotlin
internal fun getWrappedReturnType(returnType: KType): KType
return when {

   returnType.isSubclassOf(DataFetcherResult::class) -> getWrappedReturnType(returnType.getTypeOfFirstArgument())
   returnType.isSubclassOf(Publisher::class) -> { checkTypeForDataFetcherResult(returnType) }
   returnType.isSubclassOf(CompletableFuture::class) -> { checkTypeForDataFetcherResult(returnType) }
   returnType.isSubclassOf(List::class) -> {
            List::class.createType(listOf(KTypeProjection(KVariance.INVARIANT, getWrappedReturnType(returnType.getTypeOfFirstArgument()))))
        }
        else -> returnType
}
```

Reply:
I don't think this is possible without changing the existing behaviour. For example, currently for an invalid case, CompletableFuture<Publisher<String>> returns Publisher<String>; whereas the fully-recursive implementation would return String.

To be honest I am not sure why the function behaves as it does now. But because this is my first PR to the library, and because it fixes a pressing issue for us, I wanted to keep the scope of the change small, to just fix our use-case.